Signed Callbacks in Common Guide (#3770)
Nexmo/nexmo-developer: _documentation/en/concepts/guides/webhooks.md

## Decoding signed webhooks

Signed webhooks are supported by Messages, Dispatch and Voice APIs and are enabled by default. They provide a method for your application to verify a request is coming from Vonage and its payload has not been tampered with during transit. When receiving a request, the incoming webhook will include a JWT token in the authorization header which is signed with your signature secret.

Validating signed webhooks provides a number of security benefits, including:

> * Ability to verify a request originates from Vonage
> * Ensures that the message has not been tampered with while in transit
> * Defends against interception and later replay

Reviewer: ```suggestion
* The ability to verify a request originates from Vonage
* Ensuring that the message has not been tampered with while in transit
* Defending against interception and later replay
```

### Validating signed webhooks

There are two parts to validating signed webhooks:

> 1. Verify the request
> 2. Verify the payload (optional)

Reviewer: ```suggestion
1. Verifying the request
2. Verifying the payload (optional)
```

> #### Verify the request

Reviewer: ```suggestion
#### Verifying the request
```

The code example below shows how to verify a webhook signature. It is recommended you use HTTPS protocol as it ensures that the request and response are encrypted on both the client and server ends.

```code_snippets
source: '_examples/messages/signed-webhooks'
```

### Sample Signed JWT

```json
// header
{
  "alg": "HS256",
  "typ": "JWT",
}
// payload
{
  "iat": 1587494962,
  "jti": "c5ba8f24-1a14-4c10-bfdf-3fbe8ce511b5",
  "iss": "Vonage",
  "payload_hash" : "d6c0e74b5857df20e3b7e51b30c0c2a40ec73a77879b6f074ddc7a2317dd031b",
  "api_key": "a1b2c3d",
  "application_id": "aaaaaaaa-bbbb-cccc-dddd-0123456789ab"
}
```

#### Signed JWT Header

The contents of the signed JWT header are described in the following table:

Header | Value
-- | --
`alg` | HS256
`typ` | JWT

#### Signed JWT Payload

The contents of the signed JWT payload are described in the following table, using the values included in the sample signed JWT shown previously:

Field | Example Value | Description
--- | --- | ---

> `iat` |  `1587494962` | The time at which the JWT was issued. Unix timestamp in SECONDS.

Reviewer: any reason for seconds to be in caps?

Author: hm not sure... it was like this in the old guide.
